app/org/sagebionetworks/bridge/validators/StudyValidator.java

```diff
@@ -66,16 +73,39 @@ public void validate(Object obj, Errors errors) {
             errors.rejectValue("identifier", "is required");
         } else {
             if (!study.getIdentifier().matches(BridgeConstants.BRIDGE_IDENTIFIER_PATTERN)) {
-                errors.rejectValue("identifier", "must contain only lower-case letters and/or numbers with optional dashes");
+                errors.rejectValue("identifier", BRIDGE_IDENTIFIER_ERROR);
             }
             if (study.getIdentifier().length() < 2) {
                 errors.rejectValue("identifier", "must be at least 2 characters");
             }
         }
         if (study.getActivityEventKeys().stream()
-                .anyMatch(k -> !k.matches(BridgeConstants.BRIDGE_IDENTIFIER_PATTERN))) {
-            errors.rejectValue("activityEventKeys", "must contain only lower-case letters and/or numbers with " +
-                    "optional dashes");
+                .anyMatch(k -> !k.matches(BridgeConstants.SYNAPSE_IDENTIFIER_PATTERN))) {
+            errors.rejectValue("activityEventKeys", SYNAPSE_IDENTIFIER_ERROR);
+        }
+        if (study.getAutomaticCustomEvents() != null) {
+            for (Map.Entry<String, String> entry : study.getAutomaticCustomEvents().entrySet()) {
+                String key = entry.getKey();
+                String value = entry.getValue();
+                
+                // Validate that the key follows the same rules for activity event keys
+                if (!key.matches(BridgeConstants.SYNAPSE_IDENTIFIER_PATTERN)) {
+                    errors.rejectValue("automaticCustomEvents["+key+"]", SYNAPSE_IDENTIFIER_ERROR);
+                }
```

Comment:
Similarly, are we sure we're supposed to be using Synapse identifiers here?

Reply:
The regex seemed appropriate for event keys: [a-zA-Z0-9_-]. Will discuss on Slack if we want to create a different constant to avoid confusion.